### PR TITLE
Add tunable write consistency support

### PR DESCRIPTION
**Pull request by parkerduckworth**
In preparation for the v1.18 release, all write endpoint builders and their associated code have been updated to support the `consistency_level` query parameter for tunable consistency

### CODE REVIEW

File: test/batch/batchCreate_test.go [hunk 1]
```diff
@@ -158,6 +223,83 @@ func TestBatchCreate_integration(t *testing.T) {
 		testsuit.CleanUpWeaviate(t, client)
 	})
 
+	t.Run("POST /batch/references?consistency_level={level}", func(t *testing.T) {
```
Comment (dirkkul) [inline test/batch/batchCreate_test.go:226]: Could you add a short description what you're testing here? one-liner is enough
Same above

File: weaviate/batch/objectsCreate.go [hunk 2]
```diff
@@ -17,8 +18,9 @@ type ObjectsBatchRequestBody struct {
 
 // ObjectsBatcher builder to add multiple objects in one batch
 type ObjectsBatcher struct {
-	connection *connection.Connection
-	objects    []*models.Object
+	connection       *connection.Connection
+	objects          []*models.Object
+	consistencyLevel string
```
Comment (dirkkul) [inline weaviate/batch/objectsCreate.go:23]: Would it make sense to use constants instead of strings?

File: weaviate/batch/reference.go [hunk 3]
```diff
@@ -39,3 +50,11 @@ func (rb *ReferencesBatcher) Do(ctx context.Context) ([]models.BatchReferenceRes
...
 	return batchResponse, decodeErr
 }
+
+func (rb *ReferencesBatcher) buildPath() string {
+	path := "/batch/references"
+	if rb.consistencyLevel != "" {
+		path = fmt.Sprintf("%s?consistency_level=%v", path, rb.consistencyLevel)
+	}
+	return path
+}
```
Comment (dirkkul) [inline weaviate/batch/reference.go:60]: total nitpick:
I've seen this piece of code 3 times, could we somehow combine it?

File: weaviate/data/path.go [hunk 4]
```diff
@@ -40,15 +51,28 @@ func buildObjectsPath(id, className string, dbVersion *util.DBVersionSupport, ac
 	return p
 }
 
-func buildReferencesPath(id, className, referenceProperty string, dbVersion *util.DBVersionSupport) string {
-	if dbVersion.SupportsClassNameNamespacedEndpoints() {
-		if len(className) > 0 {
-			return fmt.Sprintf("/objects/%v/%v/references/%v", className, id, referenceProperty)
+// func buildReferencesPath(id, className, referenceProperty string, dbVersion *util.DBVersionSupport) string {
```
Comment (dirkkul) [inline weaviate/data/path.go:54]: remove comment?

File: weaviate/data/referenceCreator.go [hunk 5]
```diff
@@ -45,9 +46,24 @@ func (rc *ReferenceCreator) WithReference(referencePayload *models.SingleRef) *R
...
+// Can be one of 'ALL', 'ONE', or 'QUORUM'.
+func (rc *ReferenceCreator) WithConsistencyLevel(cl string) *ReferenceCreator {
+	rc.consistencyLevel = cl
+	return rc
+}
+
 // Do add the reference specified by the set payload to the object and property specified in the builder.
 func (rc *ReferenceCreator) Do(ctx context.Context) error {
-	path := buildReferencesPath(rc.uuid, rc.className, rc.referenceProperty, rc.dbVersionSupport)
+	// path := buildReferencesPath(rc.uuid, rc.className, rc.referenceProperty, rc.dbVersionSupport)
```
Comment (dirkkul) [inline weaviate/data/referenceCreator.go:59]: remove comment

File: weaviate/batch/objectsCreate.go (hunk 2 above)
Comment (parkerduckworth) [reply weaviate/batch/objectsCreate.go:23]: We have constants already in the server code, but they are not available in weaviate server version `1.17.x`. I can just upgrade to latest for now, until v1.18 is released

Comment (parkerduckworth) [reply weaviate/batch/objectsCreate.go:23]: actually, I forgot that we already had consistency level constants in this codebase. Updated to use those

File: weaviate/batch/reference.go (hunk 3 above)
Comment (parkerduckworth) [reply weaviate/batch/reference.go:60]: Refactored into its own package which handles path building across the codebase

File: test/batch/batchCreate_test.go (hunk 1 above)
Comment (parkerduckworth) [reply test/batch/batchCreate_test.go:226]: done

**Review by dirkkul — APPROVED** [review]

**Pull request merged**